### DISCUSSION TITLE
Long double tests

### DISCUSSION THREAD

**Discussion by tinko92** [boostorg/geometry#1340] [General]
When building and running the boost geometry test suite on a laptop with a Snapdragon X CPU (aarch64) in WSL (current develop e78a9fd3f576f4fc83c671b45a8f9ccbabe5bd81, GCC 14.1), I get test suite failures for 8 targets, I think, output is attached because it is rather verbose.

[test_out.txt](https://github.com/user-attachments/files/17749488/test_out.txt)

I think, most of this is related to long double being different on this architecture (or rather it is just a soft-float implementation), namely 128 bit rather than 80 bit, so this is not caught by `constexpr bool has_long_double =  sizeof(long double) > sizeof(double);` in ./test/algorithms/overlay/get_clusters.cpp or ./test/algorithms/detail/approximately_equals.cpp, at least not in the sense of guaranteeing the same behavior one might expect on x86-64. The test could be changed to std::numeric_limits<long double>::digits == 64 and added to places where it is missing. This will fix some but not all of the test case failures, so maybe those are because long double is also used in library code. I see very few spots of this happening directly in code but I don't know how many places use 
```
./include/boost/geometry/util/math.hpp:105:        using std::fabs; // for long double
./include/boost/geometry/util/math.hpp:334:struct square_root<long double, true>
./include/boost/geometry/util/math.hpp:335:    : square_root_for_fundamental_fp<long double>
./include/boost/geometry/srs/projections/str_cast.hpp:103:struct str_cast_traits_strtox<long double, false, false>
./include/boost/geometry/srs/projections/str_cast.hpp:105:    static inline long double apply(const char *str, char **str_end)
./include/boost/geometry/strategies/default_length_result.hpp:57:            long double
```

So I opted to open this as a discussion instead of a PR: 
- Should the long double tests use the criterion digits==64 instead (ruling out smaller long double types as well as more precise ones), is this true to the intention of the long double tests?
- Should long double be used within the headers at all, if results may vary significantly between platforms? I think it's 64 digits on x86-64, 113 digits on the default GCC settings for Snapdragon X and afaik 53 (same as double) with clang for Apple Silicon but I don't have a machine to test that.

**Comment by barendgehrels** [comment]
@tinko92 thanks!

On my Mac, the types `double` and `long double` are the same , meaning I personally cannot test it anymore.

Using digits==64 instead to test sounds good to me. Indeed, if things are fine then `long double` should not be used at many places. However, as a default for (for example) length result, it still sounds OK to me unless you have a better alternative.
